### PR TITLE
closes RestComm/jain-slee.sip#36

### PR DESCRIPTION
**Pull request by ThE-MaRaC**

### CODE REVIEW

File: src/gov/nist/javax/sip/stack/SIPDialog.java [hunk 1]
```diff
@@ -1914,6 +1914,7 @@ public boolean addTransaction(SIPTransaction transaction) {
                 && !firstTransactionId.equals(transaction.getBranchId())
                 && transaction.getMethod().equals(firstTransactionMethod)) {
             setReInviteFlag(true);
+            ackProcessed = false;
```
Comment (jaimecasero) [inline src/gov/nist/javax/sip/stack/SIPDialog.java]: have you tested this change somehow? would it be possible to add a junit test that test this condition?

Comment (ThE-MaRaC) [inline src/gov/nist/javax/sip/stack/SIPDialog.java]: In the bug report (https://github.com/RestComm/jain-slee.sip/issues/36) there is a SIPP TC which was used to test this correction, but I can additionally create junit TC.

Comment (jaimecasero) [inline src/gov/nist/javax/sip/stack/SIPDialog.java]: if you dont mind, adding a testCase would be great to ensure no regressions are present in the future..

File: src/test/unit/gov/nist/javax/sip/stack/reInvite/Shootist.java [hunk 2]
```diff
@@ -97,6 +97,8 @@
 
 	public boolean isTargetRefresh;
 
+	public boolean isAckWithSameBranch;
```
Comment (jaimecasero) [inline src/test/unit/gov/nist/javax/sip/stack/reInvite/Shootist.java]: i assume we expect this to be "false" by default,so it only applies to new test...

**Comment by jaimecasero** [comment]
thanks @ThE-MaRaC  for all your efforts on making this PR. I tested the test, and checked the logic. we are ready to merge!!!

**Comment by deruelle** [comment]
Thanks for the contribution. Acknowledged at https://telestax.com/acknowledgements/#jainsip

**Pull request merged**